.template/lib/utils/wrappers/permission_wrapper.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:injectable/injectable.dart';
+import 'package:permission_handler/permission_handler.dart'
+    as permission_handler;
+
+abstract class PermissionWrapper {
+  Future<bool> requestCameraPermission();
+
+  Future<bool> isCameraPermissionDenied();
+
+  Future<bool> isCameraPermissionPermanentlyDenied();
+
+  Future<bool> requestGalleryPermission();
+
+  Future<bool> isGalleryPermissionDenied();
+
+  Future<bool> isGalleryPermissionPermanentlyDenied();
+
+  Future<bool> openAppSettings();
```

Comment:
I think we just need to prepare 1 sample permission wrapping here as the base of future implementation